app/assets/javascripts/discourse/controllers/ignore-duration-with-username.js.es6

```diff
@@ -0,0 +1,31 @@
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import User from "discourse/models/user";
+
+export default Ember.Controller.extend(ModalFunctionality, {
+  loading: false,
+  ignoredUntil: null,
+  ignoredUsername: null,
+  actions: {
+    ignore() {
+      if (!this.get("ignoredUntil") || !this.get("ignoredUsername")) {
+        this.flash(
+          I18n.t("user.user_notifications.ignore_duration_time_frame_required"),
+          "alert-error"
+        );
+        return;
+      }
+      this.set("loading", true);
+      User.findByUsername(this.get("ignoredUsername")).then(user => {
+        user
+          .updateNotificationLevel("ignore", this.get("ignoredUntil"))
+          .then(() => {
+            this.get("onSuccess")(this.get("ignoredUsername"));
```

Comment:
if you are sure onSuccess is always defined here, you can do: `this.onSucess(this.get("ignoredUsername")`

The get here won’t protect you from undefined object as you call a function on it:

```
this.get("unknown")()
// Uncaught TypeError: this.get(...) is not a function
```

Reply:
I think it is less to do with `protection` from calling an undefined and more to `standari`ing` the way I access the component's properties, though I think it is okay to change it here 👍